jax/experimental/array_api/_elementwise_functions.py

```diff
@@ -229,214 +66,29 @@ def hypot(x1, x2, /):
   return jax.numpy.hypot(x1, x2)
 
 
-def imag(x, /):
-  """Returns the imaginary component of a complex number for each element x_i of the input array x."""
-  x, = _promote_dtypes("imag", x)
-  return jax.numpy.imag(x)
-
-
-def isfinite(x, /):
-  """Tests each element x_i of the input array x to determine if finite."""
-  x, = _promote_dtypes("isfinite", x)
-  return jax.numpy.isfinite(x)
-
-
-def isinf(x, /):
-  """Tests each element x_i of the input array x to determine if equal to positive or negative infinity."""
-  x, = _promote_dtypes("isinf", x)
-  return jax.numpy.isinf(x)
-
-
-def isnan(x, /):
-  """Tests each element x_i of the input array x to determine whether the element is NaN."""
-  x, = _promote_dtypes("isnan", x)
-  return jax.numpy.isnan(x)
-
-
-def less(x1, x2, /):
-  """Computes the truth value of x1_i < x2_i for each element x1_i of the input array x1 with the respective element x2_i of the input array x2."""
-  x1, x2 = _promote_dtypes("less", x1, x2)
-  return jax.numpy.less(x1, x2)
-
-
-def less_equal(x1, x2, /):
-  """Computes the truth value of x1_i <= x2_i for each element x1_i of the input array x1 with the respective element x2_i of the input array x2."""
-  x1, x2 = _promote_dtypes("less_equal", x1, x2)
-  return jax.numpy.less_equal(x1, x2)
-
-
-def log(x, /):
-  """Calculates an implementation-dependent approximation to the natural (base e) logarithm for each element x_i of the input array x."""
-  x, = _promote_dtypes("log", x)
-  return jax.numpy.log(x)
-
-
-def log1p(x, /):
-  """Calculates an implementation-dependent approximation to log(1+x), where log refers to the natural (base e) logarithm, for each element x_i of the input array x."""
-  x, = _promote_dtypes("log", x)
-  return jax.numpy.log1p(x)
-
-
-def log2(x, /):
-  """Calculates an implementation-dependent approximation to the base 2 logarithm for each element x_i of the input array x."""
-  x, = _promote_dtypes("log2", x)
-  return jax.numpy.log2(x)
-
-
-def log10(x, /):
-  """Calculates an implementation-dependent approximation to the base 10 logarithm for each element x_i of the input array x."""
-  x, = _promote_dtypes("log10", x)
-  return jax.numpy.log10(x)
-
-
-def logaddexp(x1, x2, /):
-  """Calculates the logarithm of the sum of exponentiations log(exp(x1) + exp(x2)) for each element x1_i of the input array x1 with the respective element x2_i of the input array x2."""
-  x1, x2 = _promote_dtypes("logaddexp", x1, x2)
-  return jax.numpy.logaddexp(x1, x2)
-
-
+# TODO(micky774): Update jnp.logical_* binary ops signatures to positional only
```

Comment:
I don't think this TODO is neccessary – the `jax.numpy` operations currently accept positional arguments (although they don't require them).

Reply:
Good point. At first I had done so in response to the `array-api-tests` suite, but upon further consideration it is actually a [bug](https://github.com/data-apis/array-api-tests/issues/256) in the suite 👍